src/version.c

```diff
@@ -0,0 +1,218 @@
+/* version.c
+ *
+ * Copyright (C) 2006-2023 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+#include <wolfssl/version.h>
+
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#include <wolfssl/wolfcrypt/settings.h>
+
+
+#if defined(WOLFSSL_ESPIDF)
+    #include <esp_log.h>
+    #include "sdkconfig.h"
+    const char* TAG = "Version";
+    #define WOLFSSL_VERSION_PRINTF(...) ESP_LOGI(TAG, __VA_ARGS__)
+#elif defined(WOLFSSL_NEW_PLATFORM_HERE)
+    /* define new platform-specific printf() here */
+    #define WOLFSSL_VERSION_PRINTF(...) printf(__VA_ARGS__);
+#else
+    #define WOLFSSL_VERSION_PRINTF(...) printf(__VA_ARGS__); printf("\n");
+#endif
+
+/*
+*******************************************************************************
+** Specific Platforms
+*******************************************************************************
+*/
+
+/*
+** Specific platforms: Espressif
+*/
+#if defined(WOLFSSL_ESPIDF)
+int ShowExtendedSystemInfo_platform_espressif()
+{
+#if CONFIG_IDF_TARGET_ESP32
+    WOLFSSL_VERSION_PRINTF("CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ = %u MHz",
+                            CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ
+            );
+    WOLFSSL_VERSION_PRINTF("Xthal_have_ccount = %u",
+                            Xthal_have_ccount);
+
+    #if defined(SINGLE_THREADED)
+        /* need stack monitor for single thread */
+    #else
+        WOLFSSL_VERSION_PRINTF("Stack HWM: %d",
+                               uxTaskGetStackHighWaterMark(NULL));
+    #endif
+
+#elif CONFIG_IDF_TARGET_ESP32S2
+    WOLFSSL_VERSION_PRINTF("CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ = %u MHz",
+                            CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ
+            );
+    WOLFSSL_VERSION_PRINTF("Xthal_have_ccount = %u",
+                            Xthal_have_ccount);
+#elif CONFIG_IDF_TARGET_ESP32C3
+    /* not supported at this time */
+#elif CONFIG_IDF_TARGET_ESP32S3
+    WOLFSSL_VERSION_PRINTF("CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ = %u MHz",
+                            CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ
+            );
+    WOLFSSL_VERSION_PRINTF("Xthal_have_ccount = %u",
+                            Xthal_have_ccount);
+#elif CONFIG_IDF_TARGET_ESP32H2
+    /* not supported at this time */
```

Comment:
Don't put in unnecessary conditions. When we add support we can add those targets.

Reply:
agreed. changed logic to known supported targets, everything else is an error